PORTAL/jobs/_pyperformance.py

```diff
@@ -3129,10 +3089,9 @@ def add(
         for readme in readmes:
             repo.add(readme)
         msg = f'Add Benchmark Results ({info.uploadid.copy(suite=None)})'
-        repo.commit(msg)
-        logger.info('...done committing')
-
         if push:
+            repo.commit(msg)
+            logger.info('...done committing')
```

Comment:
This is just to help with testing -- `git commit` requires that a git user and email is set up, which it isn't in CI.  It seems fine to not commit if we aren't going to push and tie these two concepts together, but this could also be a new `commit` flag (but `push = True` would require `commit` to be `True`).

Reply:
FWIW, I was doing commit-without-push in some cases.  It was mostly to work around issues with GitHub auth, but I expect that might still be needed occasionally.